include/float_vector.h

```diff
@@ -45,7 +52,7 @@ namespace quda {
     return make_double4(x.x + y.x, x.y + y.y, x.z + y.z, x.w + y.w);
   }
 
-  __host__ __device__ inline float4 operator*(const float a, const float4 x) {
+  __host__ __device__ inline float4 operator*(const float &a, const float4 &x) {
```

Comment:
what's the motivation for making `a` a reference here? Shouldn't passing by value be perfectly fine here?